pytest_splunk_addon/standard_lib/sample_generation/pytest_splunk_addon_data_parser.py

```diff
@@ -0,0 +1,160 @@
+#
+# Copyright 2021 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+import re
+import logging
+from .rule import raise_warning
+from . import SampleStanza
+
+import addonfactory_splunk_conf_parser_lib as conf_parser
+
+LOGGER = logging.getLogger("pytest-splunk-addon")
+
+
+PSA_DATA_CONFIG_FILE = "pytest-splunk-addon-data.conf"
+
+
+class PytestSplunkAddonDataParser:
+    """
+    This class parses pytest-splunk-addon-data.conf file.
+
+    Args:
+        addon_path: Path to the Splunk App
+    """
+
+    conf_name = " "
+
+    def __init__(self, addon_path: str, config_path: str):
+        self._conf_parser = conf_parser.TABConfigParser()
+        self.config_path = config_path
+        self._psa_data = None
+        self.addon_path = addon_path
+        self.match_stanzas = set()
+
+    def _path_to_samples(self):
+        if os.path.exists(os.path.join(self.config_path, "samples")):
+            LOGGER.info(
+                "Samples path is: {}".format(os.path.join(self.config_path, "samples"))
+            )
+            return os.path.join(self.config_path, "samples")
+        elif os.path.exists(
+            os.path.join(
+                os.path.abspath(os.path.join(self.config_path, os.pardir)), "samples"
+            )
+        ):
+            LOGGER.info(
+                "Samples path is: {}".format(
+                    os.path.join(
+                        os.path.abspath(os.path.join(self.config_path, os.pardir)),
+                        "samples",
+                    )
+                )
+            )
+            return os.path.join(
+                os.path.abspath(os.path.join(self.config_path, os.pardir)), "samples"
+            )
+        else:
+            LOGGER.info(
+                "Samples path is: {}".format(os.path.join(self.addon_path, "samples"))
+            )
+            return os.path.join(self.addon_path, "samples")
+
+    @property
+    def psa_data(self):
+        psa_data_path = os.path.join(self.config_path, PSA_DATA_CONFIG_FILE)
+        if os.path.exists(psa_data_path):
+            self._conf_parser.read(psa_data_path)
+            self.conf_name = "psa-data-gen"
+            self._psa_data = self._conf_parser.item_dict()
+            return self._psa_data
+        else:
+            LOGGER.warning(f"{PSA_DATA_CONFIG_FILE} not found")
+            raise FileNotFoundError(f"{PSA_DATA_CONFIG_FILE} not found")
```

Comment:
Will it be a problem for new addons which are missing `pytest-splunk-addon-data.conf`? We don't want failing tests for them

Reply:
Add-ons should not miss `pytest-splunk-addon-data.conf`, before I checked and every add-on has this one.

Reply:
New addons generated by `addonfactory-repository-template` miss it. Maybe adding empty `pytest-splunk-addon-data.conf` would be a solution?

Reply:
Yep, this may be a solution, I'll need to double check.